engine/src/main/java/org/hibernate/validator/constraintvalidation/HibernateConstraintValidatorInitializationContext.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.constraintvalidation;
+
+import java.time.Clock;
+import java.time.Duration;
+
+import javax.validation.ClockProvider;
+
+import org.hibernate.validator.Incubating;
+import org.hibernate.validator.spi.scripting.ScriptEvaluator;
+import org.hibernate.validator.spi.scripting.ScriptEvaluatorFactory;
+import org.hibernate.validator.spi.scripting.ScriptEvaluatorNotFoundException;
+
+/**
+ * Provides contextual data and operations when initializing constraint validator.
+ *
+ * @author Marko Bekhta
+ * @since 6.1
+ */
+@Incubating
+public interface HibernateConstraintValidatorInitializationContext {
+
+	/**
+	 * Returns a {@link ScriptEvaluator} created based on the {@link ScriptEvaluatorFactory}
+	 * passed at bootstrap.
+	 *
+	 * @param languageName the name of the scripting language
+	 *
+	 * @return a script executor for the given language. Never null.
+	 *
+	 * @throws ScriptEvaluatorNotFoundException in case no {@link ScriptEvaluator} was
+	 * found for a given {@code languageName}
+	 */
+
+	ScriptEvaluator getScriptEvaluatorForLanguage(String languageName);
+
+	/**
+	 * Returns clock skew tolerance as {@link Duration} which is used to determine
+	 * acceptable margin of error in milliseconds, which is allowed
+	 * when comparing date/time in time related constraints.
+	 *
+	 * @return a tolerance as  {@link Duration}
+	 */
+	Duration getClockSkewTolerance();
+
+	/**
+	 * Returns the provider for obtaining the current time in the form of a {@link Clock},
+	 * e.g. when validating the {@code Future} and {@code Past} constraints.
+	 *
+	 * @return the provider for obtaining the current time, never {@code null}. If no
+	 * specific provider has been configured during bootstrap, a default implementation using
+	 * the current system time and the current default time zone as returned by
+	 * {@link Clock#systemDefaultZone()} will be returned.
+	 */
+	ClockProvider getClockProvider();
```

Comment:
hmm, my understanding was that `ClockProvider` got into `ConstraintValidatorContext` only because there was no other way to pass it to validators at init time. I was even thinking about caching `Clock` in validator during init rather than a provider ...

Reply:
Hmmm. It sure is appealing to move the `ClockProvider` to the initialization context and caching the `Clock` would be a good idea in this case.

But I don't think we would be able to move it to the initialization context while following the spec.

And I'm not even sure we would be able to find a way to change the spec to support it in the future. As we would still have a `ClockProvider` in the `ConstraintValidatorContext`.

@gunnarmorling what's your take on this one?

Reply:
I'm not following on the caching idea. You want to query what "now" is each time when validating the constraint and not when initialising it.

Reply:
Oops, sorry, was replying from wrong browser with wrong user session :)

Reply:
A `Clock` gets you a new now every time you ask for it. At least that's my understanding (expect if you use a fixed one of course).

Reply:
Interesting, I always perceived `Clock` as a representation of a single instant, but it seems I was wrong. Yeah, caching would work then. It seems not like something though we could do right now.